sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_connection.py

```diff
@@ -507,10 +510,13 @@ def _incoming_end(self, channel, frame):
         """
         try:
             self._incoming_endpoints[channel]._incoming_end(frame)  # pylint:disable=protected-access
+            self._incoming_endpoints.pop(channel)
+            self._outgoing_endpoints.pop(channel)
         except KeyError:
-            pass  # TODO: channel error
-        #self._incoming_endpoints.pop(channel)  # TODO
-        #self._outgoing_endpoints.pop(channel)  # TODO
+            end_error = AMQPError(condition=ErrorCondition.InvalidField, description=f"Invalid channel {channel}", info=None)
```

Comment:
Is InvalidField the best error condition here?

Reply:
Changed it to `ClientError` since if a channel is missing from the dictionary (either when the frame comes from the client itself or the service) its something the SDK didn't handle on close properly.